### PR TITLE
Add SIG Node cherry-pick guide

### PR DESCRIPTION
**Pull request by ehashman**
/kind documentation
/sig node

/cc @SergeyKanzhelev @hasheddan

### CODE REVIEW

File: contributors/devel/sig-node/cherry-picks.md [hunk 1]
```diff
@@ -0,0 +1,115 @@
+# SIG Node Cherry-Pick Management Process
+
+Kubernetes supports multiple release versions simultaneously. How do new fixes
+make it into earlier versions, when we only develop one version at a time?
+
+We [cherry-pick] them!
+
+[cherry-pick]: https://git-scm.com/docs/git-cherry-pick
```
Comment (SergeyKanzhelev) [inline contributors/devel/sig-node/cherry-picks.md:8]: not the best link. May be misleading. Maybe https://github.com/kubernetes/community/blob/master/contributors/devel/sig-release/cherry-picks.md?

File: contributors/devel/sig-node/cherry-picks.md [hunk 2]
```diff
@@ -0,0 +1,115 @@
+# SIG Node Cherry-Pick Management Process
+
+Kubernetes supports multiple release versions simultaneously. How do new fixes
```
Comment (SergeyKanzhelev) [inline contributors/devel/sig-node/cherry-picks.md:3]:
```suggestion
Kubernetes supports [multiple release versions](https://kubernetes.io/docs/setup/release/version-skew-policy/) simultaneously. How do new fixes
```

File: contributors/devel/sig-node/cherry-picks.md [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+process.
+
+The week of the cherry-pick deadline, create a Slack thread ([example
+thread]) in the [#sig-node] channel to track the cherry-picks.
+
+To determine what fixes need to be cherry-picked, you can use the [SIG Node PR
+Board with the cherry-pick filter view]. Track each change considered for
+inclusion individually.
+
+Release branches are typically kept in sync. This means that bugfixes should be
```
Comment (SergeyKanzhelev) [inline contributors/devel/sig-node/cherry-picks.md:45]: should we require to support in all versions higher than minimal version for cherry-pick?

File: contributors/devel/sig-node/cherry-picks.md [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+Throughout the release cycle, merged bugfixes should be evaluated for inclusion
+as cherry-picks. Any bugs marked **critical-urgent** should be considered. Some
+**important-soon** bugs may also be considered if they are sufficiently
+contained and meet the criteria above. **Only** bug fixes should be considered for backports.
+
+To create a cherry-pick PR, you can use the `./hack/cherry_pick_pull.sh`
+script. Read through the [cherry pick documentation] for details on that
+process.
+
+The week of the cherry-pick deadline, create a Slack thread ([example
```
Comment (SergeyKanzhelev) [inline contributors/devel/sig-node/cherry-picks.md:38]: I'd say we need to have a process for before the last week. Let's mention that for questionable cherry-picks, those can be discussed in Slack or in SIG Node meeting

File: contributors/devel/sig-node/cherry-picks.md [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+[patch releases]: https://github.com/kubernetes/sig-release/blob/master/release-engineering/versioning.md#patch-releases
+[target dates]: https://github.com/kubernetes/sig-release/blob/master/releases/patch-releases.md#upcoming-monthly-releases
+[triaged]: triage.md
+
+## Monthly cherry-pick process
+
+Throughout the release cycle, merged bugfixes should be evaluated for inclusion
+as cherry-picks. Any bugs marked **critical-urgent** should be considered. Some
+**important-soon** bugs may also be considered if they are sufficiently
+contained and meet the criteria above. **Only** bug fixes should be considered for backports.
```
Comment (SergeyKanzhelev) [inline contributors/devel/sig-node/cherry-picks.md:32]: I would propose to require some soak time for the fix in master. Maybe merged for a week without known regressions in tests

File: contributors/devel/sig-node/cherry-picks.md (hunk 1 above)
Comment (ehashman) [reply contributors/devel/sig-node/cherry-picks.md:8]: I already link that. None of the k8s docs explain what a "cherry-pick" is so I think it's valuable to include for context.

File: contributors/devel/sig-node/cherry-picks.md (hunk 5 above)
Comment (ehashman) [reply contributors/devel/sig-node/cherry-picks.md:32]: Yes, this is true, I should add that.

**Review by hasheddan — COMMENTED** [review]
@ehashman It feels like a lot of this information is present in https://github.com/kubernetes/community/blob/master/contributors/devel/sig-release/cherry-picks.md. Do you feel like there is value in having a separate document for sig-node? Could parts you think are missing be added to that doc or could this one be simplified and point at that one? My main concern is just having fragmented instructions for how folks actually go through the CP process :)

File: contributors/devel/sig-node/cherry-picks.md (hunk 1 above)
Comment (SergeyKanzhelev) [reply contributors/devel/sig-node/cherry-picks.md:8]: same concept, but not the same implementation. I just don't want to confuse anybody.

**Comment by ehashman** [comment]
@hasheddan I tried to keep the duplication to the minimum. My main goal was to document the SIG-specific PR wrangling (e.g. how to actually check and surface each set of cherry-pick PRs) which isn't really written down anywhere.

File: contributors/devel/sig-node/cherry-picks.md (hunk 3 above)
Comment (ehashman) [reply contributors/devel/sig-node/cherry-picks.md:45]: Yes, that's the goal of having supported versions. All supported branches receive support.

File: contributors/devel/sig-node/cherry-picks.md (hunk 4 above)
Comment (ehashman) [reply contributors/devel/sig-node/cherry-picks.md:38]: The target audience of this doc is the Node person stewarding the cherry-picks, not authors proposing cherry-picks. I agree that people should raise controversial cherry-picks early, but it's not the responsibility of the cherry-pick steward.

**Comment by ehashman** [comment]
/label tide/merge-method-squash

So you can see the diff more easily, I'll push separate commits.

File: contributors/devel/sig-node/cherry-picks.md (hunk 4 above)
Comment (ehashman) [reply contributors/devel/sig-node/cherry-picks.md:46]: Added a paragraph making this audience/purpose more clear.

**Comment by SergeyKanzhelev** [comment]
/lgtm


follow up: https://github.com/kubernetes/community/pull/5706

**Review by kikisdeliveryservice — COMMENTED** [review]
If we're talking about cherry-picks, maybe mention here or in the triage file how to categorize them in the triage board? (I get a little confused :smile: )

Add: happy to update the triage file if you'd like, but just lmk what category you'd want those in.

**Comment by dchen1107** [comment]
Wow, we have our own cherry pick guidance now. Thanks!

/lgtm
/approve

**Pull request merged**